### PR TITLE
Algorithms t3  MKobets

### PR DESCRIPTION
**Pull request by kolya-kobets**

### CODE REVIEW

File: algorithms task (3)/student/include/impl.h [hunk 1]
```diff
@@ -13,4 +15,14 @@
  * @note behavior is undefined if shift out of bounds
  */
 template<class BiIt>
-BiIt shift_right_n(BiIt begin, BiIt end, typename BiIt::difference_type shift);
+BiIt shift_right_n(BiIt begin, BiIt end, typename BiIt::difference_type shift)
+{
+    ///Didn't get the idea for "end" parameter, as shift specifies the number of elements to shift.
```
Comment (smay1613) [inline algorithms task (3)/student/include/impl.h:20]: Task will be changed, thanks

File: algorithms task (3)/student/include/impl.h [hunk 2]
```diff
@@ -13,4 +15,14 @@
...
+    ///begin+shift - start of the destination sequence
+    //return std::move_backward(begin, std::next(begin, shift), std::next(begin, shift*2));
+
+    //source & dest ranges never overlap, so move can be used instead of move_backward.
```
Comment (smay1613) [inline algorithms task (3)/student/include/impl.h:24]: Why are you sure that source and destination will not overlap?

Comment (kolya-kobets) [reply algorithms task (3)/student/include/impl.h:24]: shift - specifies the number of elements to move
begin - source
begin+shift - destination

Comment (smay1613) [reply algorithms task (3)/student/include/impl.h:24]: range:
1   2   3   4   5
^   .   ^ 
b   .    e

n = 1

result:
1 1 2 4 5

don't they overlap?

Comment (kolya-kobets) [reply algorithms task (3)/student/include/impl.h:24]: as far as I understood, end is ignored for the task. range to move is specified by  (begin, split) pair.

Comment (smay1613) [reply algorithms task (3)/student/include/impl.h:24]: we shift the subrange

**Pull request closed without merging**